DNN Platform/DotNetNuke.Abstractions/Application/IHostSettings.cs

```diff
@@ -0,0 +1,321 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+namespace DotNetNuke.Abstractions.Application;
+
+using System;
+
+using DotNetNuke.Abstractions.Security;
+
+/// <summary>Provides access to well-known host settings.</summary>
+public interface IHostSettings
+{
+    /// <summary>Gets the duration before an account is automatically unlocked.</summary>
+    public TimeSpan AutoAccountUnlockDuration { get; }
+
+    /// <summary>Gets the <c>HttpCacheability</c> value for authenticated requests.</summary>
+    /// <remarks>The following mapping is used:
+    /// <list type="bullet">
+    /// <item><c>"0"</c> - <c>HttpCacheability.NoCache</c></item>
+    /// <item><c>"1"</c> - <c>HttpCacheability.Private</c></item>
+    /// <item><c>"2"</c> - <c>HttpCacheability.Public</c></item>
+    /// <item><c>"3"</c> - <c>HttpCacheability.Server</c></item>
+    /// <item><c>"4"</c> - <c>HttpCacheability.ServerAndNoCache</c></item>
+    /// <item><c>"5"</c> - <c>HttpCacheability.ServerAndPrivate</c></item>
+    /// </list>
+    /// </remarks>
+    public string AuthenticatedCacheability { get; }
+
+    /// <summary>Gets the <c>HttpCacheability</c> value for authenticated requests.</summary>
+    /// <remarks>The following mapping is used:
+    /// <list type="bullet">
+    /// <item><c>"0"</c> - <c>HttpCacheability.NoCache</c></item>
+    /// <item><c>"1"</c> - <c>HttpCacheability.Private</c></item>
+    /// <item><c>"2"</c> - <c>HttpCacheability.Public</c></item>
+    /// <item><c>"3"</c> - <c>HttpCacheability.Server</c></item>
+    /// <item><c>"4"</c> - <c>HttpCacheability.ServerAndNoCache</c></item>
+    /// <item><c>"5"</c> - <c>HttpCacheability.ServerAndPrivate</c></item>
+    /// </list>
+    /// </remarks>
+    public string UnauthenticatedCacheability { get; }

```

Comment:
It feels wrong to have a string representing an enum, since these values are fixed and used here:

https://github.com/dnnsoftware/Dnn.Platform/blob/db46a7dcb5e1981855d9c838664bf6bac4e36fef/DNN%20Platform/Website/Default.aspx.cs#L430-L436

We cannot use `System.Web.HttpCacheability` in `DotNetNuke.Abstractions`, since this is not available in .NET Standard 2.0 (only .NET FX).

Would it be better to create a new enum in `DotNetNuke.Abstractions` and return this here?

Reply:
Yeah, we can do that 👍🏻